src/format/opamTypes.mli

```diff
@@ -273,21 +284,14 @@ type user_action =
 
 (** Solver universe *)
 type universe = {
+  u_pefuniv  : ((string * string), Opam.Packages.package) Hashtbl.t;
+  u_options  : (string * string list * string list);
+  u_tables   : Pef.Pefcudf.tables;
```

Comment:
We can substitute these with sets/maps . They are not supposed to be mutable. On the other side I do many Hashtbl.find, so I think that a O(1) data structure makes sense here. (in particular for the tables/version_map ...